packages/core/rfcs/agnostic-address.md

```diff
@@ -0,0 +1,25 @@
+# Agnostic address type
+
+## Motivation
+
+Sometimes we  need to have access to the address property, such as street name, postcode, country etc.
+That is very useful for instance if you want to format these data and display in  the way you want.
+
+```ts
+interface AgnosticAddress {
+  streetName?: string;
+  postcode?: string;
+  city?: string:
+  streetNumber?: string;
+  firstName?: string;
+  lastName?: string;
+  phone?: string;
+  company?: string;
+  region?: string;
+  title?: string;
+}
+```
+
+## Migration process
+
+Prety straightforward, there is no address used besides comemrcools orders.
```

Comment:
in my opinion, the address isn't agnostic, like Product, every platform may have different implementations, shopware example

```ts
export interface BillingAddress {
  countryId: string;
  salutationId: string;
  street: string;
  zipcode: string;
  city: string;
  title?: string;
  additionalAddressLine1?: string;
  additionalAddressLine2?: string;
  phoneNumber?: string;
}

export interface ShippingAddress extends BillingAddress {
  firstName: string;
  lastName: string;
}
```

Reply:
still, you can map your fields  into the AgnosticAddress

countryId -> country,
salutationId -> region,
street -> street,
zipcode -> postcode,
city -> city,
title -> title,
additionalAddressLine1, additionalAddressLine2 -> additionalAddres (we can add it)
phoneNumber  -> number